### ISSUE TITLE
Show encoding progress

### ISSUE THREAD

**Issue by kparal** [EasyScreenCast/EasyScreenCast#102]
When the recording is demanding (can't be done in real time), the video is encoded even after the recording has stopped. So e.g. you record 30 seconds of your screen, then stop the recording, and the file is still being created for another 60 seconds. That is not quite obvious and it confused me at first - if you run the video immediately, it's incomplete, if you wait, it gets complete.

If it is technically possible, you can stow a dialog with encoding progress and hide it once the encoding is complete? Then it would be obvious that the video is not immediately useable, but you need to wait until it is finished.

Thanks.
## Steps to Reproduce (for bugs)
1. record something that is demanding for your computer (large screen, high fps, slow encoder)
2. stop the recording, but see that the file is still being created for some time (the file size goes up, cpu utilization shows gnome-shell >= 100% cpu)
3. wait until the video is complete (how can you tell? is it obvious?)
## Your Environment
- where you downloaded the extension?
  [x] Gnome shell extension website
  [ ] Last release from git repo
  [ ] Last version from git on master branch repo
- gnome shell version:
  gnome-shell-3.21.90.1-1.fc25.x86_64
- Operating System and version:
  Fedora 25

**Comment by iacopodeenosee** [comment]
> If it is technically possible, you can stow a dialog with encoding progress and hide it once the encoding is complete? Then it would be obvious that the video is not immediately useable, but you need to wait until it is finished.

i can not monitoring the encoding process, I don't know which is, but i think is possible **monitoring the size of the screencast file**; when the file has stopped growing i assume that the action of enconding is finished, if you have better ideas, let me know. thx

**Comment by kparal** [comment]
I hoped you're executing some gst command to record the screen, and the process exits only after the full video has been encoded. Is that not the case? Does the process exit before the video is fully saved?

**Comment by iacopodeenosee** [comment]
nope, this extension use the internal recorder of gnome shell, it pass the various arguments to gnome-shell by dbus, I don't know which gnome-shell thread manages the video encoding.

**Comment by kparal** [comment]
Ah, in that case, I understand there's little you can do about this (except for filing an RFE for gnome to publish this progress information over dbus, or at least a finish notification). I don't know how reliable file size monitoring would be (or maybe if you can test whether the file is being opened for writing) - if it is something you have time to play with, it would be definitely better than nothing.

**Comment by iacopodeenosee** [comment]
**NOTE: monitoring screencast file to catch the ending of the encoding process**
